nabs/suspenders.py

```diff
@@ -0,0 +1,179 @@
+import asyncio
+import copy
+import logging
+from threading import RLock, Event
+
+from bluesky.plan_stubs import null, wait_for
+from bluesky.preprocessors import plan_mutator
+
+logger = logging.getLogger(__name__)
+
+
+class SuspendPreprocessor:
+    """
+    Base preprocessor that suspends on specific commands based on a signal.
+
+    Parameters
+    ----------
+    signal: ``Signal``
+        The signal to subscribe to, whose value determines when to suspend.
+
+    commands: ``list of str``, optional
+        The commands to suspend on. If omitted, we'll suspend on all commands.
```

Comment:
Is there commands we never want to `suspend` on? For instance I think we are always find "saving"? We are always fine "dropping"?

Reply:
You are probably correct, we can omit some commands by default, e.g. any commands that don't interact with devices.